### PR TITLE
add: custom auto

### PR DESCRIPTION
**Pull request by chloedia**
- Add a custom auto logic that is based on the proportion of images

### CODE REVIEW

File: benchmark/process_time.py [hunk 1]
```diff
@@ -49,5 +55,5 @@ async def test_process_folder(folder_path, api_key):
 if __name__ == "__main__":
     api_key = os.getenv("MEGAPARSE_API_KEY")
     # folder_path = "megaparse/sdk/examples/only_pdfs"
-    folder_path = "/Users/amine/data/quivr/only_pdfs/"
```
Comment (Abh-iab) [inline benchmark/process_time.py:52]: folder_path =" /using/amine/data/quivr/only_pdfs/"

File: libs/megaparse/src/megaparse/parser/base.py [hunk 2]
```diff
@@ -11,6 +11,7 @@ async def convert(
         self,
         file_path: str | Path | None = None,
         file: IO[bytes] | None = None,
+        file_extensions: str = "",
```
[the remark this answers is not in the document]
Comment (chloedia) [reply libs/megaparse/src/megaparse/parser/base.py:14]: yes, it is automatically retrieved in megaparse.load / aload

File: libs/megaparse/src/megaparse/parser/llama.py [hunk 3]
```diff
@@ -30,6 +30,7 @@ async def convert(
         self,
         file_path: str | Path | None = None,
         file: IO[bytes] | None = None,
+        file_extensions: str = "",
```
[the remark this answers is not in the document]
Comment (chloedia) [reply libs/megaparse/src/megaparse/parser/llama.py:33]: same

File: libs/megaparse/src/megaparse/parser/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AmineDiro) [reply libs/megaparse/src/megaparse/parser/base.py:14]:
```suggestion
        file_extensions: str | FileExtension = "",
```

**Review by AmineDiro — APPROVED** [review]

**Pull request merged**